### PR TITLE
gcs: OSX: set flag to prevent application nap.

### PR DESCRIPTION
**Pull request by mlyle**
as discovered by @PTDreamer.  Supersedes #2064.

### CODE REVIEW

**Comment by mlyle** [comment]
Needs more verification.  Appears to do the right thing.

**Comment by peabody124** [comment]
I tested this and left GCS in the background. Came back an hour later it wasn't locked up. This is a fabulous improvement.

**Comment by peabody124** [comment]
@PTDreamer ok to merge with you? think works for me.

File: ground/gcs/src/app/main.cpp [hunk 1]
```diff
@@ -263,8 +263,15 @@ int main(int argc, char **argv)
         // fix Mac OS X 10.9 (mavericks) font issue
         // https://bugreports.qt-project.org/browse/QTBUG-32789
         QFont::insertSubstitution(".Lucida Grande UI", "Lucida Grande");
+
+        QSettings tmpSettings;
+
+        bool value = true;
+
```
Comment (kubark42) [inline ground/gcs/src/app/main.cpp:270]: I'd like to see documentation here. This code is opaque and hard to maintain code if there's no description about what it does.

**Comment by kubark42** [comment]
> I tested this and left GCS in the background. Came back an hour later it wasn't locked up. This is a fabulous improvement.

Awesome!

File: ground/gcs/src/app/main.cpp [hunk 2]
```diff
@@ -263,8 +263,15 @@ int main(int argc, char **argv)
...
+        tmpSettings.setValue(QString("NSAppSleepDisabled"), value);
```
Comment (PTDreamer) [inline ground/gcs/src/app/main.cpp:271]: Why use 3 different lines to do this? Wouldn't:

```
QSettings().setValue(QString("NSAppSleepDisabled"), true);
```

accomplish the same thing?

**Comment by PTDreamer** [comment]
> @PTDreamer ok to merge with you? think works for me.

I would prefer the more enigmatic way of doing this I proposed but hey (jk)
I made a small comment but I don't mind merging as is.

**Comment by PTDreamer** [comment]
Well this is unfortunate...

**Comment by peabody124** [comment]
Let's just reopen this and merge as is. It works, is tested. No need to quibble the details.

**Comment by PTDreamer** [comment]
Humm that will be difficult with the source branch deleted, the same happened with other PRs. Fortunately this one is easy to reproduce

**Pull request closed without merging**